src/basic/util.h

```diff
@@ -147,6 +147,13 @@ static inline int memcmp_safe(const void *s1, const void *s2, size_t n) {
         return memcmp(s1, s2, n);
 }
 
+/* Compare s1 (length n1) with s2 (length n2). If the first bytes all match
+ * but lengths are different, then the shorter one is considered smaller. */
+static inline int memcmp_nn(const void *s1, size_t n1, const void *s2, size_t n2) {
+        return memcmp(s1, s2, MIN(n1, n2))
+            ?: CMP(n1, n2);
+}
```

Comment:
btw, this is how "lexicographic ordering" is defined, might make sense to mention that in the comment

Reply:
Of course! I was thinking of it, not sure why didn't pick those words... Thanks!